realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -166,7 +167,20 @@ public void setColumnObject(AllTypes value) {
 
     @Override
     public RealmList<AllTypes> getColumnRealmList() {
-        return new RealmList<AllTypes>(AllTypes.class, row.getLinkList(INDEX_COLUMNREALMLIST), realm);
+        // use the cached value if available
+        if (columnRealmListRealmList != null) {
+            return columnRealmListRealmList;
+        } else {
+            LinkView linkView = row.getLinkList(INDEX_COLUMNREALMLIST);
+            if (linkView == null) {
+                // return empty non managed RealmList if the LinkView is null
+                // useful for non-initialized RealmObject (async query return empty Row while the query is performing)
+                return new RealmList<AllTypes>();
```

Comment:
This means that:

```
Foo foo = realm.where(Foo.class).findFirstAsync();
assertTrue(foo.getColumnRealmList() != foo.getColumnRealmResult());
```

This seems kind strange. I would prefer that 
```
assertTrue(foo.getColumnRealmList() == foo.getColumnRealmResult());
```

So perhaps replace this with `Collections.emptyList()`